src/polyfem/State.cpp

```diff
@@ -1102,66 +1104,115 @@ namespace polyfem
 		Eigen::MatrixXd node_positions;
 		Eigen::MatrixXi boundary_edges, boundary_triangles;
 		std::vector<Eigen::Triplet<double>> displacement_map_entries;
-		io::OutGeometryData::extract_boundary_mesh(*mesh, n_bases, bases, total_local_boundary,
-												   node_positions, boundary_edges, boundary_triangles, displacement_map_entries);
+		io::OutGeometryData::extract_boundary_mesh(
+			*mesh, n_bases, bases, total_local_boundary, node_positions,
+			boundary_edges, boundary_triangles, displacement_map_entries);
+		// assert(node_positions.rows() == n_bases);
 
-		Eigen::VectorXi codimensional_nodes;
-		if (obstacle.n_vertices() > 0)
+		// n_bases already contains the obstacle vertices
+		const int num_fe_nodes = node_positions.rows() - obstacle.n_vertices();
+
+		Eigen::MatrixXd collision_vertices;
+		Eigen::VectorXi collision_codim_vids;
+		Eigen::MatrixXi collision_edges, collision_triangles;
+
+		if (args.contains("/contact/collision_mesh"_json_pointer)
+			&& args.at("/contact/collision_mesh/enabled"_json_pointer).get<bool>())
```

Comment:
this reader should have an unit test, there are strange byte sizes differences between windows and unix